# **Superstore Sales and Profit Analysis**



## Dataset

This project uses the **Sample Superstore** dataset, a popular retail sales dataset containing order-level details such as sales, profit, discounts, customer information, regional breakdowns, and product categories.  
The dataset includes fields like `Order Date`, `Region`, `Category`, `Sales`, `Profit`, `Discount`, and more.

---

## Objective

The main goal of this project is to analyze sales and profit performance across different regions, segments, and product categories using **SQL queries in Python (via pandasql)**.  
By deriving actionable business insights, this analysis aims to help the company:

- Identify top-performing regions and customer segments
- Understand which product categories drive profitability
- Assess the impact of discount strategies on profits
- Support data-driven marketing and operational decision-making

---

## Business Value

This project showcases how SQL and Python can be combined to perform practical business analytics, making it valuable for:

- Guiding marketing strategy (e.g., where to focus campaigns)
- Informing pricing and discount policies
- Improving customer targeting and retention
- Supporting strategic decisions with data-backed evidence

---

## Tools Used

- Python (pandas, pandasql)
- Google Colab (for interactive analysis)

---

## Outcome

The final output includes SQL-based insights, visualizations, and business recommendations that can help the Superstore business improve its overall performance and profitability.



In [25]:
# Installing library for SQL support
!pip install pandasql

In [26]:
import pandas as pd

df = pd.read_csv('/content/Sample - Superstore.csv', encoding='latin1')

In [27]:
df.head()

Row ID        Order ID  Order Date   Ship Date       Ship Mode Customer ID  \
0       1  CA-2016-152156   11/8/2016  11/11/2016    Second Class    CG-12520   
1       2  CA-2016-152156   11/8/2016  11/11/2016    Second Class    CG-12520   
2       3  CA-2016-138688   6/12/2016   6/16/2016    Second Class    DV-13045   
3       4  US-2015-108966  10/11/2015  10/18/2015  Standard Class    SO-20335   
4       5  US-2015-108966  10/11/2015  10/18/2015  Standard Class    SO-20335   

     Customer Name    Segment        Country             City  ...  \
0      Claire Gute   Consumer  United States        Henderson  ...   
1      Claire Gute   Consumer  United States        Henderson  ...   
2  Darrin Van Huff  Corporate  United States      Los Angeles  ...   
3   Sean O'Donnell   Consumer  United States  Fort Lauderdale  ...   
4   Sean O'Donnell   Consumer  United States  Fort Lauderdale  ...   

  Postal Code  Region       Product ID         Category Sub-Category  \
0       42420   South  FUR-BO-10001798        Furniture    Bookcases   
1       42420   South  FUR-CH-10000454        Furniture       Chairs   
2       90036    West  OFF-LA-10000240  Office Supplies       Labels   
3       33311   South  FUR-TA-10000577        Furniture       Tables   
4       33311   South  OFF-ST-10000760  Office Supplies      Storage   

                                        Product Name     Sales  Quantity  \
0                  Bush Somerset Collection Bookcase  261.9600         2   
1  Hon Deluxe Fabric Upholstered Stacking Chairs,...  731.9400         3   
2  Self-Adhesive Address Labels for Typewriters b...   14.6200         2   
3      Bretford CR4500 Series Slim Rectangular Table  957.5775         5   
4                     Eldon Fold 'N Roll Cart System   22.3680         2   

   Discount    Profit  
0      0.00   41.9136  
1      0.00  219.5820  
2      0.00    6.8714  
3      0.45 -383.0310  
4      0.20    2.5164  

[5 rows x 21 columns]

In [28]:
df.columns

Index(['Row ID', 'Order ID', 'Order Date', 'Ship Date', 'Ship Mode',
       'Customer ID', 'Customer Name', 'Segment', 'Country', 'City', 'State',
       'Postal Code', 'Region', 'Product ID', 'Category', 'Sub-Category',
       'Product Name', 'Sales', 'Quantity', 'Discount', 'Profit'],
      dtype='object')

In [29]:
for col in df.columns:
    print(f"--- {col} ---")
    print(df[col].unique()[:10])  # Showing first 10 unique values to avoid huge output
    print()

--- Row ID ---
[ 1  2  3  4  5  6  7  8  9 10]

--- Order ID ---
['CA-2016-152156' 'CA-2016-138688' 'US-2015-108966' 'CA-2014-115812'
 'CA-2017-114412' 'CA-2016-161389' 'US-2015-118983' 'CA-2014-105893'
 'CA-2014-167164' 'CA-2014-143336']

--- Order Date ---
['11/8/2016' '6/12/2016' '10/11/2015' '6/9/2014' '4/15/2017' '12/5/2016'
 '11/22/2015' '11/11/2014' '5/13/2014' '8/27/2014']

--- Ship Date ---
['11/11/2016' '6/16/2016' '10/18/2015' '6/14/2014' '4/20/2017'
 '12/10/2016' '11/26/2015' '11/18/2014' '5/15/2014' '9/1/2014']

--- Ship Mode ---
['Second Class' 'Standard Class' 'First Class' 'Same Day']

--- Customer ID ---
['CG-12520' 'DV-13045' 'SO-20335' 'BH-11710' 'AA-10480' 'IM-15070'
 'HP-14815' 'PK-19075' 'AG-10270' 'ZD-21925']

--- Customer Name ---
['Claire Gute' 'Darrin Van Huff' "Sean O'Donnell" 'Brosina Hoffman'
 'Andrew Allen' 'Irene Maddox' 'Harold Pawlan' 'Pete Kriz'
 'Alejandro Grove' 'Zuschuss Donatelli']

--- Segment ---
['Consumer' 'Corporate' 'Home Office']

--- Countr

In [30]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 9994 entries, 0 to 9993
Data columns (total 21 columns):
 #   Column         Non-Null Count  Dtype  
---  ------         --------------  -----  
 0   Row ID         9994 non-null   int64  
 1   Order ID       9994 non-null   object 
 2   Order Date     9994 non-null   object 
 3   Ship Date      9994 non-null   object 
 4   Ship Mode      9994 non-null   object 
 5   Customer ID    9994 non-null   object 
 6   Customer Name  9994 non-null   object 
 7   Segment        9994 non-null   object 
 8   Country        9994 non-null   object 
 9   City           9994 non-null   object 
 10  State          9994 non-null   object 
 11  Postal Code    9994 non-null   int64  
 12  Region         9994 non-null   object 
 13  Product ID     9994 non-null   object 
 14  Category       9994 non-null   object 
 15  Sub-Category   9994 non-null   object 
 16  Product Name   9994 non-null   object 
 17  Sales          9994 non-null   float64
 18  Quantity

In [31]:
df.describe()

Row ID   Postal Code         Sales     Quantity     Discount  \
count  9994.000000   9994.000000   9994.000000  9994.000000  9994.000000   
mean   4997.500000  55190.379428    229.858001     3.789574     0.156203   
std    2885.163629  32063.693350    623.245101     2.225110     0.206452   
min       1.000000   1040.000000      0.444000     1.000000     0.000000   
25%    2499.250000  23223.000000     17.280000     2.000000     0.000000   
50%    4997.500000  56430.500000     54.490000     3.000000     0.200000   
75%    7495.750000  90008.000000    209.940000     5.000000     0.200000   
max    9994.000000  99301.000000  22638.480000    14.000000     0.800000   

            Profit  
count  9994.000000  
mean     28.656896  
std     234.260108  
min   -6599.978000  
25%       1.728750  
50%       8.666500  
75%      29.364000  
max    8399.976000

In [32]:
df.isnull().sum()

Row ID           0
Order ID         0
Order Date       0
Ship Date        0
Ship Mode        0
Customer ID      0
Customer Name    0
Segment          0
Country          0
City             0
State            0
Postal Code      0
Region           0
Product ID       0
Category         0
Sub-Category     0
Product Name     0
Sales            0
Quantity         0
Discount         0
Profit           0
dtype: int64

In [33]:
df.columns = (
    df.columns
    .str.strip()                 # Removeing leading/trailing spaces
    .str.replace(' ', '_')      # Replacing spaces with underscores
    .str.replace('-', '_')      # Replacing hyphens with underscores (if any)
)

In [34]:
from pandasql import sqldf
pysqldf = lambda q: sqldf(q, globals())

# 1. Total sales and profit by region

In [35]:
query = """
SELECT Region,
       SUM(Sales) AS Total_Sales,
       SUM(Profit) AS Total_Profit
FROM df
GROUP BY Region
ORDER BY Total_Sales DESC
"""
result = pysqldf(query)
result

Region  Total_Sales  Total_Profit
0     West  725457.8245   108418.4489
1     East  678781.2400    91522.7800
2  Central  501239.8908    39706.3625
3    South  391721.9050    46749.4303

**Business Interpretation:** <br>
The West region has the highest total sales and also the highest profit, which shows that this market is performing very strongly and is a key revenue driver.
The East region is also doing well but has slightly lower profits than the West, suggesting there might be higher costs or more discounting.
The Central and South regions have significantly lower sales and profits, indicating they might need more marketing focus, better pricing strategies, or operational improvements.
Overall, the business should prioritize the West and East regions for further investment and growth initiatives, while reviewing what is holding back performance in the Central and South.

# 2. Sales and profit by product category

In [36]:
query1 = """
SELECT Category,
       SUM(Sales) AS Total_Sales,
       SUM(Profit) AS Total_Profit
FROM df
GROUP BY Category
ORDER BY Total_Sales DESC
"""
result = pysqldf(query1)
result

Category  Total_Sales  Total_Profit
0       Technology  836154.0330   145454.9481
1        Furniture  741999.7953    18451.2728
2  Office Supplies  719047.0320   122490.8008

# **Business Interpretation:** <br>
Technology products have the highest sales and also deliver the highest profit, making them the strongest overall category for the business.
Furniture, although it has high sales, shows very low profit. This might be due to higher costs, heavy discounting, or lower margin products — so the business should look into optimizing pricing or cost structures for this category.
Office Supplies have slightly lower sales than Furniture but generate a much higher profit, showing they are more efficient in terms of profitability.
Overall, the company should continue to focus on Technology and Office Supplies for growth and consider revisiting the Furniture strategy to improve profit margins.

# 3. Top 10 customers by total sales

In [19]:
query2 = """
SELECT Customer_Name,
       SUM(Sales) AS Total_Sales
FROM df
GROUP BY Customer_Name
ORDER BY Total_Sales DESC
LIMIT 10
"""
result = pysqldf(query2)
result

Customer_Name  Total_Sales
0         Sean Miller    25043.050
1        Tamara Chand    19052.218
2        Raymond Buch    15117.339
3        Tom Ashbrook    14595.620
4       Adrian Barton    14473.571
5        Ken Lonsdale    14175.229
6        Sanjit Chand    14142.334
7        Hunter Lopez    12873.298
8        Sanjit Engle    12209.438
9  Christopher Conant    12129.072

# Business Interpretation: <br>
These top 10 customers represent a significant share of the company’s total revenue, with Sean Miller standing out as the most valuable individual customer.
This highlights the importance of building and maintaining strong relationships with key customers through loyalty programs, personalized offers, or dedicated account support.
By focusing on these high-value customers, the business can increase repeat purchases, strengthen brand loyalty, and secure a more stable revenue stream.
Additionally, studying these customers' buying patterns could help identify opportunities to upsell or cross-sell other products to similar customer segments.

# 4. Yearly sales and profit trend

In [21]:
df['Order_Date'] = pd.to_datetime(df['Order_Date'])

In [22]:
query = """
SELECT strftime('%Y', Order_Date) AS Year,
       SUM(Sales) AS Total_Sales,
       SUM(Profit) AS Total_Profit
FROM df
GROUP BY Year
ORDER BY Year
"""
result = pysqldf(query)
result

Year  Total_Sales  Total_Profit
0  2014  484247.4981    49543.9741
1  2015  470532.5090    61618.6037
2  2016  609205.5980    81795.1743
3  2017  733215.2552    93439.2696

### **Business Interpretation: <br>**
Overall, the business shows a positive growth trend from 2014 to 2017, both in sales and profit.
Although there was a slight dip in sales in 2015, profits still increased that year, indicating possible improvements in cost management or pricing strategy.
From 2015 onward, both sales and profit grew steadily, with 2017 reaching the highest performance.
This consistent growth suggests strong market demand and effective business strategies. The company should continue investing in scaling up while maintaining profit margins, and also analyze what worked particularly well in 2016–2017 to replicate those strategies in future years.

# 5. Sales and profit by segment and region

In [23]:
query = """
SELECT Segment,
       Region,
       SUM(Sales) AS Total_Sales,
       SUM(Profit) AS Total_Profit
FROM df
GROUP BY Segment, Region
ORDER BY Total_Sales DESC
"""
result = pysqldf(query)
result

Segment   Region  Total_Sales  Total_Profit
0      Consumer     West  362880.7730    57450.6040
1      Consumer     East  350908.1670    41190.9843
2      Consumer  Central  252031.4340     8564.0481
3     Corporate     West  225855.2745    34437.4299
4     Corporate     East  200409.3470    23622.5789
5      Consumer    South  195580.9710    26913.5728
6     Corporate  Central  157995.8128    18703.9020
7   Home Office     West  136721.7770    16530.4150
8   Home Office     East  127463.7260    26709.2168
9     Corporate    South  121885.9325    15215.2232
10  Home Office  Central   91212.6440    12438.4124
11  Home Office    South   74255.0015     4620.6343

# **Business Interpretaion:** <br>
Consumers in the West and East regions generate the highest sales and profits, showing strong demand and suggesting that these areas should remain key focus markets.
The Consumer segment in the Central region has relatively high sales but very low profit, indicating possible pricing or discounting issues that need to be reviewed.
The Corporate segment performs well overall, especially in the West and East, which suggests opportunities to further develop B2B strategies or dedicated corporate services in these regions.
The Home Office segment shows more modest sales and profits across all regions, with particularly low profit in the South. This suggests it might not be a strong focus segment for expansion in that region, and the business should evaluate whether to improve its approach or reallocate resources.
Overall, these insights support customizing marketing and sales strategies for each segment-region combination to maximize both revenue and profitability.

# 6. Discount impact on profit

In [24]:
query = """
SELECT Discount,
       AVG(Profit) AS Avg_Profit
FROM df
GROUP BY Discount
ORDER BY Discount
"""
result = pysqldf(query)
result

Discount  Avg_Profit
0       0.00   66.900292
1       0.10   96.055074
2       0.15   27.288298
3       0.20   24.702572
4       0.30  -45.679636
5       0.32  -88.560656
6       0.40 -111.927429
7       0.45 -226.646464
8       0.50 -310.703456
9       0.60  -43.077212
10      0.70  -95.874060
11      0.80 -101.796797

#**Business Interpretation: <br>**
The analysis clearly shows that as discounts increase, the average profit declines sharply, even turning negative beyond a 30% discount.
Low or no discounts (0% and 10%) have the highest average profit, suggesting that minimal discounting is most beneficial for maintaining healthy margins.
Heavy discounts of 30% and above consistently result in losses, indicating that such aggressive discounting strategies are damaging profitability.
The business should consider limiting discounts to a maximum of around 15%–20%, carefully targeting promotions only where necessary to boost sales without hurting profit margins.
Overall, a more data-driven and controlled discount policy will help protect the company’s bottom line while still attracting price-sensitive customers.